unix_integration/src/daemon.rs

```diff
@@ -281,15 +281,15 @@ async fn handle_client(
                 debug!("pam authenticate init");
                 pam_state = PamState::Step(account_id.clone());
                 cachelayer
-                    .pam_account_authenticate_step(account_id.as_str(), None)
+                    .pam_account_authenticate_step(account_id.as_str(), None, None)
```

Comment:
Shouldn't this be a new call to pam_account_authenticate_init, which then creates an auth session we bind here into pam_state? I feel like pam state should be created by and managed by the cache layer, not the caller. That way the "pam_state" can contain things like the user token (if any) and our knowledge of the credentials and what's needed. 

Then in pamAuthenticateStep you pass that pam_state back into cachelayer, and mostly, the pam_state actually has everything needed to allow the auth to proceed already and we only actually access the db or the id provider if something online needs checking.

Reply:
Ah, I like that better. Perhaps we could eliminate passing the PamData down to the pam layer this way, since it's really only needed by the daemon, and we can just keep it within the pam_state?

Reply:
The PamState would look like this?
```rust
#[derive(Debug)]
pub enum ProviderResult {
    PamPrompt(PamPrompt),
    UserToken(Option<UserToken>),
}

pub enum PamState {
    Uninitialized,
    Init(String, PamData),
    Step(ProviderResult, PamData),
}
```